### PR TITLE
Mttl 2285

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket

[Jira Ticket](https://hackney.atlassian.net/browse/MTTL-2285)

## Describe this PR

Added TF Compliance rules

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -125,20 +169,50 @@ jobs:
     steps:
       - assume-role-and-persist-workspace:
           aws-account: $AWS_ACCOUNT_PRODUCTION
-  terraform-init-and-apply-to-development:
+  terraform-init-and-plan-development:
```
Comment (LBHMGeorgieva) [inline .circleci/config.yml:172]: Can we add the plan and terraform compliance steps as part of the checks that run automatically for every commit so PR reviewers can see the result of the terraform compliance checks before approving?

File: terraform/production/terraform-compliance/dynamodb.feature [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    Given I have aws_dynamodb_table defined
+    Then it must contain tags
+    And it must contain BackupPolicy
+
+  Scenario: Ensure point in time recovery enabled
+    Given I have aws_dynamodb_table defined
+    Then it must contain point_in_time_recovery
+    And its enabled property must be true
+
+  Scenario: Ensure a maximum of 1 LSIs
```
Comment (LBHMGeorgieva) [inline terraform/production/terraform-compliance/dynamodb.feature:16]: Can we have the same check for GSI?

File: .circleci/config.yml (hunk 1 above)
Comment (humulla) [reply .circleci/config.yml:172]: Sure, will update the workflow

File: terraform/production/terraform-compliance/dynamodb.feature (hunk 2 above)
Comment (humulla) [reply terraform/production/terraform-compliance/dynamodb.feature:16]: GSI is not being implemented in this API which is why I didn't add it. Let me know if I've understood something wrong

**Review by LBHMGeorgieva — APPROVED** [review]

**Pull request merged**